pkg/service/auth_context.go

```diff
@@ -23,44 +27,125 @@ type AuthContext struct {
 	Authorization map[*config.AuthorizationConfig]interface{}
 }
 
-// Evaluate evaluates all steps of the auth pipeline (identity → metadata → policy enforcement)
-func (self *AuthContext) Evaluate() error {
-	self.Identity = make(map[*config.IdentityConfig]interface{})
-	self.Metadata = make(map[*config.MetadataConfig]interface{})
-	self.Authorization = make(map[*config.AuthorizationConfig]interface{})
-
-	// identity (authentication)
-	identityConfigs := self.API.IdentityConfigs
-	for i := range identityConfigs {
-		c := identityConfigs[i]
-		if ret, err := c.Call(self); err != nil {
+// AuthObjectConfig provides an interface for APIConfig objects that implements a Call method
+type AuthObjectConfig interface {
+	Call(ctx common.AuthContext) (interface{}, error)
+}
+
+type configCallback = func(config AuthObjectConfig, obj interface{})
+
+// NewAuthContext creates an AuthContext instance
+func NewAuthContext(parentCtx context.Context, req *auth.CheckRequest, apiConfig config.APIConfig) AuthContext {
+
+	return AuthContext{
+		ParentContext: &parentCtx,
+		Request:       req,
+		API:           &apiConfig,
+		Identity:      make(map[*config.IdentityConfig]interface{}),
+		Metadata:      make(map[*config.MetadataConfig]interface{}),
+		Authorization: make(map[*config.AuthorizationConfig]interface{}),
+	}
+
+}
+
+func (authContext *AuthContext) getAuthObject(ctx context.Context, objConfig AuthObjectConfig, cb configCallback) error {
```

Comment:
Also thinking about a better naming for "AuthObject" here 🤔 ... Not sure yet. "AuthPartial" perhaps? It could also be "AuthResponse" even though it's not the full response to the auth request just yet. IDK.